realm-jni/src/util.cpp

```diff
@@ -118,8 +118,8 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             break;
 
         case Unspecified:
-            jExceptionClass = env->FindClass("java/lang/RuntimeException");
-            message = "Unspecified exception. " + classStr;
+            jExceptionClass = env->FindClass("io/realm/exceptions/RealmError");
+            message = "Unrecoverable error condition. " + classStr;
```

Comment:
@kneth, as far as I remember, an `mmap()` failure would cause a `std::runtime_error()` to be thrown , which in turn would be categorized as `RuntimeError` rather than `Unspecified`. Isn't `Unspecified` only used for the `catch (...)` case?